// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
edition = "2023";

package koladata.ext.persisted_data;

message CompositeInitialDataManagerMetadata {
  // The version of the manager. Always set. Example format: "1.0.0"
  string version = 1;

  // Metadata about the component managers. Will always contain at least one
  // element.
  repeated ComponentManagerMetadata component_manager_metadata = 2;
}

message ComponentManagerMetadata {
  // The persistence directory of the manager.
  string persistence_dir = 1;

  // The index of the revision in the manager's revision history to use.
  int64 revision_history_index = 2;
}
